src/test/java/com/mael/payroll/transactions/AddCommissionedEmployeeTest.java

```diff
@@ -30,12 +32,12 @@ public void addsCommissionedEmployeeToTheDB() {
 
     @Test
     public void addsTheRightSalary() {
-        assertEquals(1000.0, ((CommissionedPayment)employee.getPaymentType()).getMonthlyRate(), 0.001);
+        assertEquals(1000.0, commissionedPayment.getMonthlyRate(), 0.001);
```

Comment:
:+1:
The cast in `setUp` is still a smell as you correctly pointed out, but we can address that later.